m1well.Utils/src/utilsHelper.js

```diff
@@ -0,0 +1,202 @@
+// @flow
+
+import { isFuture, isMatch, parse, startOfTomorrow } from 'date-fns'
+import { showMessage } from '../../helpers/userInput'
+
+const DATE_FORMAT_ISO = 'yyyy-MM-dd'
+
+export type Config = {
+  archiveNotesTag: string,
+  archiveNotesLifeInDays: number,
+}
+
+/**
+ * cast string from the config mixed
+ *
+ * @param val the config mixed
+ * @param key name of the property you want to cast
+ * @returns {string} casted value
+ */
+export const castStringFromMixed = (val: { [string]: ?mixed }, key: string): string => {
+  const result = findPropertyInMixed(val, key)
+  return result ? ((result: any): string) : ''
+}
+
+/**
+ * cast number from the config mixed
+ *
+ * @param val the config mixed
+ * @param key name of the property you want to cast
+ * @returns {string} casted value
+ */
+export const castNumberFromMixed = (val: { [string]: ?mixed }, key: string): number => {
+  const result = findPropertyInMixed(val, key)
+  return result ? ((result: any): number) : 0
+}
+
+/**
+ * Currently there is no function to create a daily note in the future, if it doesn't exist
+ * Hopefully this is coming very soon - then add this here to this function
+ *
+ * @param day future date
+ * @returns {?TNote} note if it exists
+ */
+export const getDailyNote = (day: Date): ?TNote => {
+  return DataStore.calendarNoteByDate(day)
+}
+
+/**
+ * sort function for paragraphs by prio `!!!`, `!!` and `!`
+ *
+ * @returns {(function(TParagraph, TParagraph): (number))}
+ */
+export const sortByPrio = (): Function => {
+  return (second: TParagraph, first: TParagraph) => {
+    if (haveSameTypes(first, second)) {
+      if (isFlaggedThrice(first) || isFlaggedThrice(second)) {
+        return isFlaggedThrice(first) ? 1 : -1
+      }
+      if (isFlaggedTwice(first)) {
+        return isFlaggedThrice(second) ? -1 : 1
+      }
+      if (isFlaggedTwice(second)) {
+        return isFlaggedThrice(first) ? 1 : -1
+      }
+      if (isFlaggedOnce(first)) {
+        return (isFlaggedTwice(second) || isFlaggedThrice(second)) ? -1 : 1
+      }
+      if (isFlaggedOnce(second)) {
+        return (isFlaggedTwice(second) || isFlaggedThrice(first)) ? 1 : -1
+      }
+    }
+    return 0
+  }
+}
+
+/**
+ * sort by type (open, scheduled, cancelled, done, list, text) by it's type
+ *
+ * @returns {(function(TParagraph, TParagraph): (number))} sorted paragraphs
+ */
+export const sortByType = (): Function => {
```

Comment:
I find this super complicatated. I'm sure it works and you've tested it, but I am sure there's a shorter way to do this.

Reply:
i don't know how - I just played around with these custom sort functions and this works as expected - there is even a test for it ;)